src/radio/index.es

```diff
@@ -76,23 +76,27 @@ class Radio extends InputComponent {
   }
 
   render() {
-    const { type, options, children } = this.props;
+    const { type, options, children, label } = this.props;
     const className = clsx('c-radio', this.props.className, {
       [`is-type-${type}`]: type,
       'is-error': this.isErrorActive(),
     });
 
+    let labelNode;
+    if (label) labelNode = <legend className="ui-label ui-strong">{label}</legend>;
+
     const radios = options.map(this.renderRadio);
 
     let error;
     if (this.isErrorActive()) error = <em className="ui-error">{this.getError()}</em>;
 
     return (
-      <article className={className}>
+      <fieldset className={className}>
```

Comment:
Fieldset is semantically correct but can be hard to style. 

normalize.css helps with that therefore it should be ok imo

Tested in Chrome, FF and Safari (most recent versions)